centralizedmetrics/providers/appsflyer.go

```diff
@@ -0,0 +1,122 @@
+package providers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/status-im/status-go/centralizedmetrics/common"
+)
+
+const AppsflyerBaseURL = "https://api3.appsflyer.com"
+
+var AppsflyerAppID = ""
+var AppsflyerToken = ""
+
+// AppsflyerMetricProcessor implements MetricProcessor for Appsflyer
+type AppsflyerMetricProcessor struct {
+	AppID   string
+	Secret  string
+	BaseURL string
+}
+
+// NewAppsflyerMetricProcessor is a constructor for AppsflyerMetricProcessor
+func NewAppsflyerMetricProcessor(appID, secret, baseURL string) *AppsflyerMetricProcessor {
+	return &AppsflyerMetricProcessor{
+		AppID:   appID,
+		Secret:  secret,
+		BaseURL: baseURL,
+	}
+}
+
+func (amp *AppsflyerMetricProcessor) GetAppID() string {
+	if len(amp.AppID) != 0 {
+		return amp.AppID
+	}
+	return AppsflyerAppID
+}
+
+func (amp *AppsflyerMetricProcessor) GetToken() string {
+	if len(amp.Secret) != 0 {
+		return amp.Secret
+	}
+
+	return AppsflyerToken
+}
+
+// Process processes an array of metrics and sends them to the Appsflyer API
+func (amp *AppsflyerMetricProcessor) Process(metrics []common.Metric) error {
+	for _, metric := range metrics {
+		if err := amp.sendToAppsflyer(metric); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// sendToAppsflyer sends a single metric to the Appsflyer API
+func (amp *AppsflyerMetricProcessor) sendToAppsflyer(metric common.Metric) error {
+	url := fmt.Sprintf("%s/inappevent/%s", amp.BaseURL, amp.GetAppID())
+
+	payload, err := json.Marshal(toAppsflyerMetric(metric))
+	if err != nil {
+		return err
+	}
+
+	log.Info("sending metrics to", "url", url, "metric", metric, "secret", amp.GetToken())
```

Comment:
Could you elaborate why we need to log the token?

Reply:
oh we don't sorry, for testing :)